src/modules/colorPicker/ColorPickerUI/Settings/SettingItem.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace ColorPicker.Settings
 {
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1649:File name should match first type name", Justification = "It does, it is a generic.  False positive")]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1649:File name should match first type name", Justification = "Generic, file is named correctly")]
     public sealed class SettingItem<T> : INotifyPropertyChanged
```

Comment:
I believe the guidance is to name it SettingItem{T}.cs, as per:  https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/Configuration.md#file-naming-conventions